system/system_monitor/src/net_monitor/net_monitor.cpp

```diff
@@ -492,5 +502,121 @@ void NetMonitor::monitorTraffic(diagnostic_updater::DiagnosticStatusWrapper & st
+
+  int whole_level = DiagStatus::OK;
+  std::string error_str;
+  uint64_t total_reassembles_failed = 0;
+  uint64_t unit_reassembles_failed = 0;
+
+  if (getReassemblesFailed(total_reassembles_failed)) {
+    reassembles_failed_queue_.push_back(total_reassembles_failed - last_reassembles_failed_);
+    while (reassembles_failed_queue_.size() > reassembles_failed_check_duration_) {
+      reassembles_failed_queue_.pop_front();
+    }
+
+    for (auto reassembles_failed : reassembles_failed_queue_) {
+      unit_reassembles_failed += reassembles_failed;
+    }
+
+    stat.add(fmt::format("total packet reassembles failed"), total_reassembles_failed);
+    stat.add(fmt::format("packet reassembles failed per unit time"), unit_reassembles_failed);
+
+    if (unit_reassembles_failed >= reassembles_failed_check_count_) {
+      whole_level = std::max(whole_level, static_cast<int>(DiagStatus::WARN));
+      error_str = "reassembles failed";
+    }
+
+    last_reassembles_failed_ = total_reassembles_failed;
+  } else {
+    reassembles_failed_queue_.push_back(0);
+    whole_level = std::max(whole_level, static_cast<int>(DiagStatus::ERROR));
+    error_str = "failed to read /proc/net/snmp";
+  }
+
+  if (!error_str.empty()) {
+    stat.summary(whole_level, error_str);
+  } else {
+    stat.summary(whole_level, "OK");
+  }
+
+  // Measure elapsed time since start time and report
+  SystemMonitorUtility::stopMeasurement(t_start, stat);
+}
+
+void NetMonitor::searchReassemblesFailedColumnIndex()
+{
+  std::ifstream ifs("/proc/net/snmp");
+  if (!ifs) {
+    return;
+  }
+
+  // /proc/net/snmp
+  // Ip: Forwarding DefaultTTL InReceives ... ReasmTimeout ReasmReqds ReasmOKs ReasmFails ...
+  // Ip: 2          64         5636471397 ... 135          2303339    216166   270        ...
+  std::string line;
+
+  // Find column index of 'ReasmFails'
+  if (!std::getline(ifs, line)) {
+    return;
+  }
+
+  std::vector<std::string> title_list;
+  boost::split(title_list, line, boost::is_space());
+
+  if (title_list.empty()) {
+    return;
+  }
+  if (title_list[0] != "Ip:") {
+    return;
+  }
+
+  int index = 0;
+  for (auto itr = title_list.begin(); itr != title_list.end(); ++itr, ++index) {
+    if (*itr == "ReasmFails") {
+      reassembles_failed_column_index_ = index;
+      break;
+    }
+  }
+}
+
+bool NetMonitor::getReassemblesFailed(uint64_t & reassembles_failed)
+{
+  if (reassembles_failed_column_index_ < 0) {
+    return false;
+  }
+
+  std::ifstream ifs("/proc/net/snmp");
+  if (!ifs) {
+    return false;
+  }
+
+  std::string line;
+
+  // Skip title row
+  if (!std::getline(ifs, line)) {
+    return false;
+  }
+
+  // Find a value of 'ReasmFails'
+  if (!std::getline(ifs, line)) {
+    return false;
+  }
+
+  std::vector<std::string> value_list;
+  boost::split(value_list, line, boost::is_space());
+
+  if (value_list.size() <= static_cast<std::size_t>(reassembles_failed_column_index_)) {
```

Comment:
This conditional statement is to check if the index is more than the size of value list.
Please make it understandable easily.
```cpp
if (reassembles_failed_column_index_ >= value_list.size())
```

Reply:
@ito-san 
I fixed as pointed out.
ef247d72f3cebf0ae6cf21935dc4d86543f161f1